### PR TITLE
panc: function file_contents: support reading (gzip) compressed file

### PR DESCRIPTION
**Pull request by stdweird**
Extend the `file_contents` function to support reading gzipped files.

Allows to access gzipped data produced outside quattor without need to uncompress it first.

### CODE REVIEW

**Comment by jrha** [comment]
AFAICT the code seems fine, it's Java. :man_shrugging:

Now for some bike-shedding :bike: around naming:
* Strictly speaking gzip is a file-format — it wraps a DEFLATE compressed stream with a header and footer.
* Given that the mime-type for a gzipped file is `application/gzip` should the option be `gzip` rather than `gz`.

**Comment by jrha** [comment]
I have added an example description to this PR, please fill in the blanks :smile:

**Comment by stdweird** [comment]
@jrha renamed option to `gzip`. 
wrt descrition: it's intended to read gzipped files. i'm not sure what is missing from the description

**Comment by jrha** [comment]
What's missing is why you would want to do this, that's the important piece of information for all changes, i.e. what was the motivation to do this work?

File: panc/src/test/java/org/quattor/pan/dml/functions/FileContentsTest.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public void testTooFewArguments() throws SyntaxException {
 	@Test(expected = SyntaxException.class)
 	public void testTooManyArguments() throws SyntaxException {
 		FileContents.getInstance(null, StringProperty.getInstance("OK"),
-				StringProperty.getInstance("OK"));
+                                 StringProperty.getInstance("OK"), StringProperty.getInstance("gz"));
```
Comment (jrha) [inline panc/src/test/java/org/quattor/pan/dml/functions/FileContentsTest.java]: Should this now be `"gzip"`? Or is it unrelated to the option?

**Comment by stdweird** [comment]
@jrha both updated

**Comment by jrha** [comment]
Thanks!

**Comment by stdweird** [comment]
@jrha i don't know. at least for output eg `xml.gz` and `jons.gz` don't set this.

**Comment by jrha** [comment]
I've tested it locally by adding a unicode ellipsis to `ok.txt.gz` and it seems to work fine with or without passing `"UTF-8"` to the outer `InputStreamReader`, so I'm happy.

**Pull request merged**